### ISSUE TITLE
draw_mermaid() can't accept input other than 'str'

### ISSUE THREAD

**Issue by gbaian10** [langchain-ai/langgraph#692]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the [LangGraph](https://langchain-ai.github.io/langgraph/)/LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangGraph/LangChain rather than my code.
- [X] I am sure this is better as an issue [rather than a GitHub discussion](https://github.com/langchain-ai/langgraph/discussions/new/choose), since this is a LangGraph bug and not a design question.

### Example Code

```python
from typing import Annotated, TypedDict

from langchain_core.messages import AnyMessage
from langgraph.graph import END, START, StateGraph
from langgraph.graph.message import add_messages


class State(TypedDict):
    messages: Annotated[list[AnyMessage], add_messages]


def branch(state: State) -> bool:
    return 1 + 1 == 3


graph_builder = StateGraph(State)
graph_builder.add_node("foo", lambda state: {"messages": [("ai", "foo")]})
graph_builder.add_node("bar", lambda state: {"messages": [("ai", "bar")]})

graph_builder.add_conditional_edges(
    START,
    branch,
    path_map={True: "foo", False: "bar"},
    then=END,
)

app = graph_builder.compile()
print(app.invoke({"messages": []}))  # can work

# GOOD, But only 'True' is displayed on the path
app.get_graph().draw_png(output_file_path="graph.png")

# ERROR
print(app.get_graph().draw_mermaid())
```

### Error Message and Stack Trace (if applicable)

```shell
File "/home/gbaian10/.local/lib/python3.10/site-packages/langchain_core/runnables/graph.py", line 423, in draw_mermaid
    return draw_mermaid(
  File "/home/gbaian10/.local/lib/python3.10/site-packages/langchain_core/runnables/graph_mermaid.py", line 84, in draw_mermaid
    words = edge_data.split()  # Split the string into words
AttributeError: 'bool' object has no attribute 'split'
```

### Description

This question seems like a follow-up to #620.  The graph can work.

But mermaid_syntax can't be generated. Therefore mermaid_png can't be generated either.

In addition, in draw_png(), it only displays True and does not display False.

If change part of the above code to the following, everything will work fine. But this seems to defeat the purpose of using bool

```python
def branch(state: State) -> str:
    return str(1 + 1 == 3)

graph_builder.add_conditional_edges(
    START,
    branch,
    path_map={"True": "foo", "False": "bar"},
    then=END,
)
```

### System Info

langgraph==0.0.69
langchain==0.2.5
python_version==3.12.4
platform==linux

**Comment by vbarda** [comment]
this should be fixed in `langchain_core == 0.2.10`. let me know if you still run into issues, thanks for reporting!

**Comment by gbaian10** [comment]
> this should be fixed in `langchain_core == 0.2.10`. let me know if you still run into issues, thanks for reporting!

It can works, thank you. 😀
